### PR TITLE
Feature/processtree

### PR DESCRIPTION
**Pull request by amitschendel**

### CODE REVIEW

File: pkg/processmanager/v1/process_manager.go [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+
+// PopulateInitialProcesses scans the /proc filesystem to build the initial process tree
+// for all registered container shim processes. It establishes parent-child relationships
+// between processes and adds them to the process tree if they are descendants of a shim.
+func (p *ProcessManager) PopulateInitialProcesses() error {
+	if len(p.containerIdToShimPid.Keys()) == 0 {
+		return nil
+	}
+
+	fs, err := procfs.NewFS("/proc")
```
Comment (matthyx) [inline pkg/processmanager/v1/process_manager.go:48]: shouldn't you open `/host/proc` instead?

Comment (matthyx) [reply pkg/processmanager/v1/process_manager.go:48]: actually I see that you open `/proc` in the rule_manager as well

**Review by matthyx — DISMISSED** [review]
just check comments regarding proc

File: pkg/processmanager/v1/process_manager.go (hunk 1 above)
Comment (amitschendel) [reply pkg/processmanager/v1/process_manager.go:48]: Hmm you might be right.

Comment (amitschendel) [reply pkg/processmanager/v1/process_manager.go:48]: image
Nope see this.

Comment (matthyx) [reply pkg/processmanager/v1/process_manager.go:48]: sababa

**Review by matthyx — APPROVED** [review]

**Review by matthyx — DISMISSED** [review]

**Review by matthyx — APPROVED** [review]
looks good, thanks!

**Pull request merged**